src/app/cli/src/coda.ml

```diff
@@ -420,13 +459,42 @@ let daemon logger =
          let addrs_and_ports : Node_addrs_and_ports.t =
            {external_ip; bind_ip; peer= None; client_port; libp2p_port}
          in
+         let block_production_key =
+           maybe_from_config YJ.Util.to_string_option "block-producer-key"
+             block_production_key
+         in
+         let block_production_pubkey =
+           maybe_from_config json_to_publickey_compressed_option
+             "block-producer-pubkey" block_production_pubkey
+         in
+         let block_production_password =
+           maybe_from_config YJ.Util.to_string_option "block-producer-password"
+             block_production_password
+         in
+         Option.iter
+           ~f:(fun password ->
+             Unix.putenv ~key:Secrets.Keypair.env ~data:password )
+           block_production_password ;
```

Comment:
could you exit error or something if they have both the env var set and the password in the config file? With instructions on how to disambiguate -- would hate to confuse someone here

Reply:
Ok to log error instead of exiting?